### PR TITLE
Move offline update to api usage

### PR DESCRIPTION
**Pull request by mike-sul**
- Move the `aklite-offline` utility to the api usage since it supports offline update now.
- Remove the offline update specific client/functionality that we have used until now.
- Move the offline update test suite to testing of the api-based offline update implementation.
- Fix some other issue and improve related functionality along the way.

### CODE REVIEW

**Comment by mike-sul** [comment]
@detsch I think it is ready for review. I may add another commit to fix another tiny issue which won't impact the given PR change much.

File: apps/aklite-offline/cmds.cpp [hunk 1]
```diff
@@ -2,166 +2,139 @@
...
+  const LocalUpdateSource local_update_source{.tuf_repo = (src_dir / "tuf").string(),
+                                              .ostree_repo = (src_dir / "ostree_repo").string(),
+                                              .app_store = (src_dir / "apps").string()};
+  auto ret_code{aklite::cli::Install(client, -1, "", "", force_downgrade, &local_update_source)};
+  switch (ret_code) {
+    case aklite::cli::StatusCode::InstallAppsNeedFinalization: {
+      // TBD: The former `aklite-offline` sets `10` as a an exit/status code, while
+      // the current version returns `InstallAppsNeedFinalization = 105`.
+      // Maybe it makes sense to override it with `10`, but the `10` is already used for `TufMetaPullFailure = 10`?
+      std::cout << "Please run `aklite-offline run` command to start the updated Apps\n";
```
Comment (detsch) [inline apps/aklite-offline/cmds.cpp:29]: IMO it is ok to change the value now to make it unique globally. Perhaps @caiotpereira can give some feedback on the matter.

**Review by detsch — APPROVED** [review]
Required changes in the generic code make sense, and the code looks good.
Did some tests with my factory, the commands are working as expected.

File: apps/aklite-offline/cmds.cpp (hunk 1 above)
Comment (mike-sul) [reply apps/aklite-offline/cmds.cpp:29]: ok. I'll start discussion with Caio offline.

**Comment by mike-sul** [comment]
> @detsch I think it is ready for review. I may add another commit to fix another tiny issue which won't impact the given PR change much.

Let's merge it now, it is already has too many commits. I'll address the issues I mentioned earlier in another PR.

**Pull request merged**